src/libcore/option.rs

```diff
@@ -201,7 +201,8 @@ impl<T> Option<T> {
     /// ```
     ///
     /// [`None`]: #variant.None
-    #[must_use]
+    #[must_use = "if you intended to assert that this doesn't have a value, consider \
+                  `.and_then(|| panic!(\"`Option` had a value when expected `None`\"))` instead"]
```

Comment:
I think this message is too long.  How about the suggestion is `.ok_or(()).unwrap_err()`?

Reply:
I don't have a strong opinion; I just wanted to give a message as nice as the `.unwrap()` ones do.

Maybe there's a way to combine them? `.ok_or("Option::None").unwrap_err()` or similar?

Reply:
Can we just tell the user to wrap it in an `assert!()`?

Reply:
Really we should [add an `expect_none`](https://internals.rust-lang.org/t/adding-option-expect-none/10481).

Reply:
See #62596.